### ISSUE TITLE
Update agent on Win10 via OCO

### ISSUE THREAD

**Issue by jyevon** [schorschii/OCO-Agent#17]
I'm trying to update the OCO Agent on our Windows 10 machines via OCO (i.e. while running), using the command from your [entry in the Install/Uninstall Command Catalog](https://github.com/schorschii/OCO-Server/blob/master/docs/install-uninstall-catalog/OCO%20Agent%20Update.md) over in the server's repo. However, the installer always fails with code 5 as some files are in use (see log below).

I've also tried to stop the service of the old agent version as part of the installation command (`net stop "OCO-Agent" & oco-agent.exe ...`). However, the installation terminates with it and on the admin page the job never returns from execution state.

Similarly, it makes no sense to me to uninstall the old version first, because then there's nothing left on the machine to start the installation of the new version.

I'd appreciate some advice. Is there something I'm missing?

PS: The same scenario works flawlessly on a Linux Mint machine using the .deb package.

Package in OCO Server:
![grafik](https://github.com/schorschii/OCO-Agent/assets/54958084/3da07d9f-f288-43dc-a3e4-b8bfc696b6e6)

Installer log:
```
﻿2024-06-15 14:53:17.141   Log opened. (Time zone: UTC+02:00)
2024-06-15 14:53:17.141   Setup version: Inno Setup version 6.2.2
2024-06-15 14:53:17.141   Original Setup EXE: C:\WINDOWS\TEMP\oco-staging\oco-agent-1.1.0.exe
2024-06-15 14:53:17.141   Setup command line: /SL5="$A00A8,11441097,832512,C:\WINDOWS\TEMP\oco-staging\oco-agent-1.1.0.exe" /SILENT /SUPPRESSMSGBOXES /LOG="install.log" 
2024-06-15 14:53:17.141   Windows version: 10.0.19045  (NT platform: Yes)
2024-06-15 14:53:17.141   64-bit Windows: Yes
2024-06-15 14:53:17.141   Processor architecture: x64
2024-06-15 14:53:17.141   User privileges: Administrative
2024-06-15 14:53:17.157   Administrative install mode: Yes
2024-06-15 14:53:17.157   Install mode root key: HKEY_LOCAL_MACHINE
2024-06-15 14:53:17.157   64-bit install mode: No
2024-06-15 14:53:17.172   Created temporary directory: C:\WINDOWS\TEMP\is-T1QHK.tmp
2024-06-15 14:53:17.344   Starting the installation process.
2024-06-15 14:53:17.344   Directory for uninstall files: C:\Program Files\OCO Agent
2024-06-15 14:53:17.344   Will append to existing uninstall log: C:\Program Files\OCO Agent\unins000.dat
2024-06-15 14:53:17.344   -- File entry --
2024-06-15 14:53:17.344   Dest filename: C:\Program Files\OCO Agent\unins000.exe
2024-06-15 14:53:17.360   Time stamp of our file: 2024-06-15 14:53:16.532
2024-06-15 14:53:17.360   Dest file exists.
2024-06-15 14:53:17.360   Time stamp of existing file: 2023-10-14 18:25:38.699
2024-06-15 14:53:17.360   Version of our file: 51.1052.0.0
2024-06-15 14:53:17.360   Version of existing file: 51.1052.0.0
2024-06-15 14:53:17.360   Installing the file.
2024-06-15 14:53:17.360   Leaving temporary file in place for now.
2024-06-15 14:53:17.360   -- File entry --
2024-06-15 14:53:17.360   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-console-l1-1-0.dll
2024-06-15 14:53:17.360   Time stamp of our file: 2024-04-17 02:31:36.000
2024-06-15 14:53:17.360   Dest file exists.
2024-06-15 14:53:17.360   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.360   Installing the file.
2024-06-15 14:53:17.375   Successfully installed the file.
2024-06-15 14:53:17.375   -- File entry --
2024-06-15 14:53:17.375   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-datetime-l1-1-0.dll
2024-06-15 14:53:17.375   Time stamp of our file: 2024-04-17 02:31:36.000
2024-06-15 14:53:17.375   Dest file exists.
2024-06-15 14:53:17.375   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.375   Installing the file.
2024-06-15 14:53:17.375   Successfully installed the file.
2024-06-15 14:53:17.375   -- File entry --
2024-06-15 14:53:17.375   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-debug-l1-1-0.dll
2024-06-15 14:53:17.375   Time stamp of our file: 2024-04-17 02:31:40.000
2024-06-15 14:53:17.375   Dest file exists.
2024-06-15 14:53:17.375   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.375   Installing the file.
2024-06-15 14:53:17.375   Successfully installed the file.
2024-06-15 14:53:17.375   -- File entry --
2024-06-15 14:53:17.375   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-errorhandling-l1-1-0.dll
2024-06-15 14:53:17.375   Time stamp of our file: 2024-04-17 02:31:46.000
2024-06-15 14:53:17.375   Dest file exists.
2024-06-15 14:53:17.375   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.375   Installing the file.
2024-06-15 14:53:17.375   Successfully installed the file.
2024-06-15 14:53:17.375   -- File entry --
2024-06-15 14:53:17.375   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-file-l1-1-0.dll
2024-06-15 14:53:17.375   Time stamp of our file: 2024-04-17 02:31:48.000
2024-06-15 14:53:17.375   Dest file exists.
2024-06-15 14:53:17.375   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.375   Installing the file.
2024-06-15 14:53:17.375   Successfully installed the file.
2024-06-15 14:53:17.375   -- File entry --
2024-06-15 14:53:17.375   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-file-l1-2-0.dll
2024-06-15 14:53:17.375   Time stamp of our file: 2024-04-17 02:31:52.000
2024-06-15 14:53:17.375   Dest file exists.
2024-06-15 14:53:17.375   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.375   Installing the file.
2024-06-15 14:53:17.375   Successfully installed the file.
2024-06-15 14:53:17.375   -- File entry --
2024-06-15 14:53:17.375   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-file-l2-1-0.dll
2024-06-15 14:53:17.375   Time stamp of our file: 2024-04-17 02:31:54.000
2024-06-15 14:53:17.375   Dest file exists.
2024-06-15 14:53:17.375   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.375   Installing the file.
2024-06-15 14:53:17.391   Successfully installed the file.
2024-06-15 14:53:17.391   -- File entry --
2024-06-15 14:53:17.391   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-handle-l1-1-0.dll
2024-06-15 14:53:17.391   Time stamp of our file: 2024-04-17 02:31:58.000
2024-06-15 14:53:17.391   Dest file exists.
2024-06-15 14:53:17.391   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.391   Installing the file.
2024-06-15 14:53:17.391   Successfully installed the file.
2024-06-15 14:53:17.391   -- File entry --
2024-06-15 14:53:17.391   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-heap-l1-1-0.dll
2024-06-15 14:53:17.391   Time stamp of our file: 2024-04-17 02:32:00.000
2024-06-15 14:53:17.391   Dest file exists.
2024-06-15 14:53:17.391   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.391   Installing the file.
2024-06-15 14:53:17.391   Successfully installed the file.
2024-06-15 14:53:17.391   -- File entry --
2024-06-15 14:53:17.391   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-interlocked-l1-1-0.dll
2024-06-15 14:53:17.391   Time stamp of our file: 2024-04-17 02:32:04.000
2024-06-15 14:53:17.391   Dest file exists.
2024-06-15 14:53:17.391   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.391   Installing the file.
2024-06-15 14:53:17.391   Successfully installed the file.
2024-06-15 14:53:17.391   -- File entry --
2024-06-15 14:53:17.391   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-libraryloader-l1-1-0.dll
2024-06-15 14:53:17.391   Time stamp of our file: 2024-04-17 02:32:06.000
2024-06-15 14:53:17.391   Dest file exists.
2024-06-15 14:53:17.391   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.391   Installing the file.
2024-06-15 14:53:17.391   Successfully installed the file.
2024-06-15 14:53:17.391   -- File entry --
2024-06-15 14:53:17.391   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-localization-l1-2-0.dll
2024-06-15 14:53:17.391   Time stamp of our file: 2024-04-17 02:32:10.000
2024-06-15 14:53:17.391   Dest file exists.
2024-06-15 14:53:17.391   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.391   Installing the file.
2024-06-15 14:53:17.391   Successfully installed the file.
2024-06-15 14:53:17.391   -- File entry --
2024-06-15 14:53:17.391   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-memory-l1-1-0.dll
2024-06-15 14:53:17.391   Time stamp of our file: 2024-04-17 02:32:12.000
2024-06-15 14:53:17.391   Dest file exists.
2024-06-15 14:53:17.391   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.407   Installing the file.
2024-06-15 14:53:17.407   Successfully installed the file.
2024-06-15 14:53:17.407   -- File entry --
2024-06-15 14:53:17.407   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-namedpipe-l1-1-0.dll
2024-06-15 14:53:17.407   Time stamp of our file: 2024-04-17 02:32:14.000
2024-06-15 14:53:17.407   Dest file exists.
2024-06-15 14:53:17.407   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.407   Installing the file.
2024-06-15 14:53:17.407   Successfully installed the file.
2024-06-15 14:53:17.407   -- File entry --
2024-06-15 14:53:17.407   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-processenvironment-l1-1-0.dll
2024-06-15 14:53:17.407   Time stamp of our file: 2024-04-17 02:32:18.000
2024-06-15 14:53:17.407   Dest file exists.
2024-06-15 14:53:17.407   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.407   Installing the file.
2024-06-15 14:53:17.407   Successfully installed the file.
2024-06-15 14:53:17.407   -- File entry --
2024-06-15 14:53:17.407   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-processthreads-l1-1-0.dll
2024-06-15 14:53:17.407   Time stamp of our file: 2024-04-17 02:32:22.000
2024-06-15 14:53:17.407   Dest file exists.
2024-06-15 14:53:17.407   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.407   Installing the file.
2024-06-15 14:53:17.407   Successfully installed the file.
2024-06-15 14:53:17.407   -- File entry --
2024-06-15 14:53:17.407   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-processthreads-l1-1-1.dll
2024-06-15 14:53:17.407   Time stamp of our file: 2024-04-17 02:32:24.000
2024-06-15 14:53:17.407   Dest file exists.
2024-06-15 14:53:17.407   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.407   Installing the file.
2024-06-15 14:53:17.407   Successfully installed the file.
2024-06-15 14:53:17.407   -- File entry --
2024-06-15 14:53:17.407   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-profile-l1-1-0.dll
2024-06-15 14:53:17.407   Time stamp of our file: 2024-04-17 02:32:26.000
2024-06-15 14:53:17.407   Dest file exists.
2024-06-15 14:53:17.407   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.407   Installing the file.
2024-06-15 14:53:17.407   Successfully installed the file.
2024-06-15 14:53:17.422   -- File entry --
2024-06-15 14:53:17.422   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-rtlsupport-l1-1-0.dll
2024-06-15 14:53:17.422   Time stamp of our file: 2024-04-17 02:32:30.000
2024-06-15 14:53:17.422   Dest file exists.
2024-06-15 14:53:17.422   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.422   Installing the file.
2024-06-15 14:53:17.422   Successfully installed the file.
2024-06-15 14:53:17.422   -- File entry --
2024-06-15 14:53:17.422   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-string-l1-1-0.dll
2024-06-15 14:53:17.422   Time stamp of our file: 2024-04-17 02:32:34.000
2024-06-15 14:53:17.422   Dest file exists.
2024-06-15 14:53:17.422   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.422   Installing the file.
2024-06-15 14:53:17.422   Successfully installed the file.
2024-06-15 14:53:17.422   -- File entry --
2024-06-15 14:53:17.422   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-synch-l1-1-0.dll
2024-06-15 14:53:17.422   Time stamp of our file: 2024-04-17 02:32:36.000
2024-06-15 14:53:17.422   Dest file exists.
2024-06-15 14:53:17.422   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.422   Installing the file.
2024-06-15 14:53:17.422   Successfully installed the file.
2024-06-15 14:53:17.422   -- File entry --
2024-06-15 14:53:17.422   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-synch-l1-2-0.dll
2024-06-15 14:53:17.422   Time stamp of our file: 2024-04-17 02:32:40.000
2024-06-15 14:53:17.422   Dest file exists.
2024-06-15 14:53:17.422   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.422   Installing the file.
2024-06-15 14:53:17.422   Successfully installed the file.
2024-06-15 14:53:17.422   -- File entry --
2024-06-15 14:53:17.422   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-sysinfo-l1-1-0.dll
2024-06-15 14:53:17.422   Time stamp of our file: 2024-04-17 02:32:42.000
2024-06-15 14:53:17.422   Dest file exists.
2024-06-15 14:53:17.422   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.422   Installing the file.
2024-06-15 14:53:17.422   Successfully installed the file.
2024-06-15 14:53:17.422   -- File entry --
2024-06-15 14:53:17.422   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-timezone-l1-1-0.dll
2024-06-15 14:53:17.422   Time stamp of our file: 2024-04-17 02:32:46.000
2024-06-15 14:53:17.422   Dest file exists.
2024-06-15 14:53:17.422   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.422   Installing the file.
2024-06-15 14:53:17.438   Successfully installed the file.
2024-06-15 14:53:17.438   -- File entry --
2024-06-15 14:53:17.438   Dest filename: C:\Program Files\OCO Agent\api-ms-win-core-util-l1-1-0.dll
2024-06-15 14:53:17.438   Time stamp of our file: 2024-04-17 02:32:48.000
2024-06-15 14:53:17.438   Dest file exists.
2024-06-15 14:53:17.438   Time stamp of existing file: 2019-02-21 05:06:30.000
2024-06-15 14:53:17.438   Installing the file.
2024-06-15 14:53:17.438   Successfully installed the file.
2024-06-15 14:53:17.438   -- File entry --
2024-06-15 14:53:17.438   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-conio-l1-1-0.dll
2024-06-15 14:53:17.438   Time stamp of our file: 2024-04-17 02:32:52.000
2024-06-15 14:53:17.438   Dest file exists.
2024-06-15 14:53:17.438   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.438   Installing the file.
2024-06-15 14:53:17.438   Successfully installed the file.
2024-06-15 14:53:17.438   -- File entry --
2024-06-15 14:53:17.438   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-convert-l1-1-0.dll
2024-06-15 14:53:17.438   Time stamp of our file: 2024-04-17 02:32:54.000
2024-06-15 14:53:17.438   Dest file exists.
2024-06-15 14:53:17.438   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.438   Installing the file.
2024-06-15 14:53:17.438   Successfully installed the file.
2024-06-15 14:53:17.438   -- File entry --
2024-06-15 14:53:17.438   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-environment-l1-1-0.dll
2024-06-15 14:53:17.438   Time stamp of our file: 2024-04-17 02:32:58.000
2024-06-15 14:53:17.438   Dest file exists.
2024-06-15 14:53:17.438   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.438   Installing the file.
2024-06-15 14:53:17.438   Successfully installed the file.
2024-06-15 14:53:17.438   -- File entry --
2024-06-15 14:53:17.438   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-filesystem-l1-1-0.dll
2024-06-15 14:53:17.438   Time stamp of our file: 2024-04-17 02:33:00.000
2024-06-15 14:53:17.438   Dest file exists.
2024-06-15 14:53:17.438   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.438   Installing the file.
2024-06-15 14:53:17.438   Successfully installed the file.
2024-06-15 14:53:17.438   -- File entry --
2024-06-15 14:53:17.438   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-heap-l1-1-0.dll
2024-06-15 14:53:17.438   Time stamp of our file: 2024-04-17 02:33:04.000
2024-06-15 14:53:17.438   Dest file exists.
2024-06-15 14:53:17.438   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.438   Installing the file.
2024-06-15 14:53:17.454   Successfully installed the file.
2024-06-15 14:53:17.454   -- File entry --
2024-06-15 14:53:17.454   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-locale-l1-1-0.dll
2024-06-15 14:53:17.454   Time stamp of our file: 2024-04-17 02:33:06.000
2024-06-15 14:53:17.454   Dest file exists.
2024-06-15 14:53:17.454   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.454   Installing the file.
2024-06-15 14:53:17.454   Successfully installed the file.
2024-06-15 14:53:17.454   -- File entry --
2024-06-15 14:53:17.454   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-math-l1-1-0.dll
2024-06-15 14:53:17.454   Time stamp of our file: 2024-04-17 02:33:10.000
2024-06-15 14:53:17.454   Dest file exists.
2024-06-15 14:53:17.454   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.454   Installing the file.
2024-06-15 14:53:17.454   Successfully installed the file.
2024-06-15 14:53:17.454   -- File entry --
2024-06-15 14:53:17.454   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-multibyte-l1-1-0.dll
2024-06-15 14:53:17.454   Time stamp of our file: 2024-04-17 02:33:12.000
2024-06-15 14:53:17.454   Dest file exists.
2024-06-15 14:53:17.454   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.454   Installing the file.
2024-06-15 14:53:17.454   Successfully installed the file.
2024-06-15 14:53:17.454   -- File entry --
2024-06-15 14:53:17.454   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-process-l1-1-0.dll
2024-06-15 14:53:17.454   Time stamp of our file: 2024-04-17 02:33:18.000
2024-06-15 14:53:17.454   Dest file exists.
2024-06-15 14:53:17.454   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.454   Installing the file.
2024-06-15 14:53:17.454   Successfully installed the file.
2024-06-15 14:53:17.454   -- File entry --
2024-06-15 14:53:17.454   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-runtime-l1-1-0.dll
2024-06-15 14:53:17.454   Time stamp of our file: 2024-04-17 02:33:22.000
2024-06-15 14:53:17.454   Dest file exists.
2024-06-15 14:53:17.454   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.454   Installing the file.
2024-06-15 14:53:17.454   Successfully installed the file.
2024-06-15 14:53:17.454   -- File entry --
2024-06-15 14:53:17.454   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-stdio-l1-1-0.dll
2024-06-15 14:53:17.454   Time stamp of our file: 2024-04-17 02:33:24.000
2024-06-15 14:53:17.454   Dest file exists.
2024-06-15 14:53:17.454   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.454   Installing the file.
2024-06-15 14:53:17.469   Successfully installed the file.
2024-06-15 14:53:17.469   -- File entry --
2024-06-15 14:53:17.469   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-string-l1-1-0.dll
2024-06-15 14:53:17.469   Time stamp of our file: 2024-04-17 02:33:28.000
2024-06-15 14:53:17.469   Dest file exists.
2024-06-15 14:53:17.469   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.469   Installing the file.
2024-06-15 14:53:17.469   Successfully installed the file.
2024-06-15 14:53:17.469   -- File entry --
2024-06-15 14:53:17.469   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-time-l1-1-0.dll
2024-06-15 14:53:17.469   Time stamp of our file: 2024-04-17 02:33:30.000
2024-06-15 14:53:17.469   Dest file exists.
2024-06-15 14:53:17.469   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.469   Installing the file.
2024-06-15 14:53:17.469   Successfully installed the file.
2024-06-15 14:53:17.469   -- File entry --
2024-06-15 14:53:17.469   Dest filename: C:\Program Files\OCO Agent\api-ms-win-crt-utility-l1-1-0.dll
2024-06-15 14:53:17.469   Time stamp of our file: 2024-04-17 02:33:34.000
2024-06-15 14:53:17.469   Dest file exists.
2024-06-15 14:53:17.469   Time stamp of existing file: 2016-03-06 00:14:44.000
2024-06-15 14:53:17.469   Installing the file.
2024-06-15 14:53:17.469   Successfully installed the file.
2024-06-15 14:53:17.469   -- File entry --
2024-06-15 14:53:17.469   Dest filename: C:\Program Files\OCO Agent\base_library.zip
2024-06-15 14:53:17.469   Time stamp of our file: 2024-05-21 11:44:24.000
2024-06-15 14:53:17.469   Dest file exists.
2024-06-15 14:53:17.469   Time stamp of existing file: 2023-09-01 20:18:44.000
2024-06-15 14:53:17.469   Installing the file.
2024-06-15 14:53:17.500   DeleteFile: The existing file appears to be in use (32). Retrying.
2024-06-15 14:53:18.518   DeleteFile: The existing file appears to be in use (32). Retrying.
2024-06-15 14:53:19.534   DeleteFile: The existing file appears to be in use (32). Retrying.
2024-06-15 14:53:20.549   DeleteFile: The existing file appears to be in use (32). Retrying.
2024-06-15 14:53:21.565   Defaulting to Abort for suppressed message box (Abort/Retry/Ignore):
                          C:\Program Files\OCO Agent\base_library.zip
                          
                          An error occurred while trying to replace the existing file:
                          DeleteFile failed; code 32.
                          Der Prozess kann nicht auf die Datei zugreifen, da sie von einem anderen Prozess verwendet wird.
2024-06-15 14:53:21.565   User canceled the installation process.
2024-06-15 14:53:21.565   Rolling back changes.
2024-06-15 14:53:21.565   Starting the uninstallation process.
2024-06-15 14:53:21.565   Uninstallation process succeeded.
2024-06-15 14:53:23.081   Deinitializing Setup.
2024-06-15 14:53:23.081   Log closed.
```

**Comment by schorschii** [comment]
Hi,

Are you updating from agent version 1.0.2? I'm sorry to tell that this version had a bug preventing a successful self-upgrade.

In contrast to Linux, executables on Windows cannot be deleted when they are running. That's a problem when trying to implement a self-updating mechanism. The OCO agent solves this by creating a new agent program folder and renaming it afterwards.

It's correct that uninstalling or stopping the agent while updating will not work since the setup will be killed too as soon as you stop the agent service.

If you need to update many machines and a manual update is not an option, I can describe a (hacky) workaround which I wouldn't recommend normally.

**Comment by jyevon** [comment]
Thank you for your help! Yes, I was indeed upgrading from version 1.0.2 (a single machine had version 1.0.3).

It's good to know that this is fixed in agent version 1.1.0 and that it's not a fundamental issue with OCO or our setup. (I just verified on one machine by manually upgrading the agent and then re-installing via OCO.) I will upgrade the agent by hand this time, hoping that it will work via OCO for future versions.
